File: 2015/AI/AIClient_Python/src/event/JoinGameEvent.py
'''
Created on Dec 15, 2014

@author: scarriere
'''
from event.IngoingEvent import IngoingEvent
from aiclient.Singleton import Singleton
from world.World import World

class JoinGameEvent(IngoingEvent):
    
    def __init__(self):
        self.yourId = 0
        
    def execute(self):
        world = Singleton(World)
        world._joinGame(self.yourId)

    def fillArguments(self, string):
        self.yourId = int(string)
        if self.yourId == 0:
            return False
        return True

File: 2015/AI/AIClient_JAVA/src/event/JoinGameEvent.java
/* ------------------------------------------------------------------------------
 ** _________   _________      ________    _____      _____  ___________ _________
 ** \_   ___ \ /   _____/     /  _____/   /  _  \    /     \ \_   _____//   _____/
 ** /    \  \/ \_____  \     /   \  ___  /  /_\  \  /  \ /  \ |    __)_ \_____  \ 
 ** \     \____/        \    \    \_\  \/    |    \/    Y    \|        \/        \
 **  \______  /_______  /     \______  /\____|__  /\____|__  /_______  /_______  /
 **         \/        \/             \/         \/         \/        \/        \/ 
 **
 ** JoinGameEvent.java
 ** Event that happen when you join a game
 **
 ** Author: Samuel-Ricardo Carriere
 ** ------------------------------------------------------------------------------*/

package event;

import world.World;

public class JoinGameEvent extends IngoingEvent {

	private int yourId;

	@Override
	public void execute() {
		World.getInstance().joinGame(yourId);
	}

	@Override
	public boolean fillArguments(String string) {
		yourId = convertCharToNumeral(string);
		if (yourId != 0) {
			return true;
		}
		return false;
	}

}
